src/Microsoft.FSharp.Collections.ListModule.erl

```diff
@@ -1,54 +1,1195 @@
+%%
+%% Returns the sum of the results generated by applying the
+%% function to each element of the list.
+%%---------------------- *** Type *** ---------------------------%%
+%% ('T  -> ^U)  -> 'T list  -> ^U
+%%---------------------------------------------------------------%%
+sumBy(Projection,List) ->
+  sumBy(Projection,List,0).
+sumBy(_,[],Acc) ->
+  Acc;
+sumBy(Projection,[H|T],Acc) ->
+  sumBy(Projection,T,Projection(H) + Acc).
+
+
+%%---------------------- *** Name *** ---------------------------%%
+%%                          tail
+%%-------------------- *** Document *** -------------------------%%
+%%
+%% Returns the input list without the first element.
+%%---------------------- *** Type *** ---------------------------%%
+%% 'T list  -> 'T list
+%%---------------------------------------------------------------%%
+tail(List) -> lists:last(List).
+
+
+%%---------------------- *** Name *** ---------------------------%%
+%%                          toArray
+%%-------------------- *** Document *** -------------------------%%
+%%
+%% Creates an array from the given list.
+%%---------------------- *** Type *** ---------------------------%%
+%% 'T list  -> 'T []
+%%---------------------------------------------------------------%%
+% TODO : Fill with the correct argument
+%toArray(_) -> ().
+
+
+%%---------------------- *** Name *** ---------------------------%%
+%%                          toSeq
+%%-------------------- *** Document *** -------------------------%%
+%%
+%% Views the given list as a sequence.
+%%---------------------- *** Type *** ---------------------------%%
+%% 'T list  -> seq&lt;'T&gt;
+%%---------------------------------------------------------------%%
+toSeq(List) ->
+  'Microsoft.FSharp.Collections.SeqModule':ofList(List).
+
+
+%%---------------------- *** Name *** ---------------------------%%
+%%                          tryFind
+%%-------------------- *** Document *** -------------------------%%
+%%
+%% Returns the first element for which the given function
+%% returns true. Return None if no such element exists.
+%%---------------------- *** Type *** ---------------------------%%
+%% ('T  -> bool)  -> 'T list  -> 'T option
+%%---------------------------------------------------------------%%
+tryFind(_,[]) ->
+  none;
+tryFind(Pred,[H|T]) ->
+  case Pred(H) of
+    true -> {some,H};
+    false -> tryFind(Pred,T)
+  end.
+
+
+%%---------------------- *** Name *** ---------------------------%%
+%%                          tryFindIndex
+%%-------------------- *** Document *** -------------------------%%
+%%
+%% Returns the index of the first element in the list that
+%% satisfies the given predicate. Return None if no such
+%% element exists.
+%%---------------------- *** Type *** ---------------------------%%
+%% ('T  -> bool)  -> 'T list  -> int option
+%%---------------------------------------------------------------%%
+tryFindIndex(Pred,List) ->
+  tryFindIndex(Pred,List,0).
+tryFindIndex(_,[],_) ->
+  none;
```

Comment:
options in fez are erased to `undefined | term()` so you should return `undefined` for `None` and just the value for `Some`